### ISSUE TITLE
Support Qt

### ISSUE THREAD

**Issue by dkfellows** [cpplint/cpplint#261]
Some of us use [Qt](https://www.qt.io/), which has its own idioms. In particular, it can have things like this in class definitions:
```cpp
class Foo : public QObject
{
    Q_OBJECT
public:
    int foo();
private slots:
    double bar(bool abc);
protected signals:
    void grill(char ch);
};
```
The three lines of interest here are: `Q_OBJECT`, `private slots:` (which can have other access modifiers too) and `protected signals:` (again, other access modifiers possible). They're used for the metaobject protocol stuff that powers Qt, and the second two really ought to be indented like standard access modifiers in a class declaration.

I'd rather not just turn off all indentation checking.

**Comment by aaronliu0130** [comment]
How about any line that ends in a colon and doesn't contain "case"?

**Comment by aaronliu0130** [comment]
@dkfellows It seems like we actually support private/protected/public/signals slots. However, we don't do the same when replacing "slots" with "signals". Are you sure `protected signals` is actually supported by Qt?

**Comment by aaronliu0130** [comment]
Closing since I don't think this is actually an issue as "protected signals" seems invalid AFAIK
